README.md

```diff
@@ -1,16 +1,148 @@
+* Libraries MUST provide a means of dynamically controlling the trace global config at runtime.
+* Libraries SHOULD keep track of active spans and in memory samples based on latency/errors and 
+offer ways to access the data.
+* Because “trace context” must also be propagated between process, library MUST offer the 
+functionality that allows any transport (e.g RPC, HTTP, etc.) systems to encode/decode the “trace
+context” for placement on the wire.
+
+##### Links
+* Trace API is described [here][TraceAPI]
+* Data model is defined [here][TraceDataModel]
+
+#### Tags
+Tags are used by the Stats API to break down measurements by arbitrary metadata set in the 
+current process or propagated from a remote caller (tags are propagated through the Generic 
+Context subsystem inside a process and between process by any transport (e.g RPC, HTTP, etc.) 
+systems).
+
+The key elements of the Tags API are:
+* A tag: this is a key-value pair, where the key is a string, and the value can be one of a 64-bit
+integer, a boolean, or a string. The API allows for creating, modifying and querying objects 
+representing a tag value.
+* A set of tags (with unique keys) carried in the Generic Context. Libraries MUST provide a means 
+of manipulating the tags in the context, including adding new tags, replacing tag values, deleting
+tags, and querying the current value for a given tag key. 
+* Because tags must also be propagated between process, library MUST offer the functionality that 
+allows RPC systems to encode/decode the set of tags for placement on the wire.
+
+##### Links
+* TODO: Add links to API definition and data model.
+
+#### Stats
+The Stats API is designed to record measurements, dynamically break them down by 
+application-defined tags, and aggregate those measurements in user-defined ways. It is designed 
+to offer multiple types of aggregation (e.g. Distributions) and be efficient (all measurement 
+processing is done as a background activity); aggregating data enables reducing the overhead of 
+uploading data, while also allowing applications direct access to stats for (e.g.) load-balancing
+purposes.
+
+The key elements the API MUST provide are:
+* Defining what is to be measured (the types of data collected, and their meaning), and how data 
+will be aggregated (e.g. into a distribution, cumulative aggregation vs. deltas, etc.). Libraries
+must offer ways for customer to define Metrics that make sense for their application, and support
+a canonical set for RPC/HTTP systems.
+* Recording data - API's for recording measured values. This data is then broken down by tags 
+carried in the context (e.g. a tag can have a value that describes the current RPC service/method
+name; when RPC latency is recorded, this can be made in a generic call, without having to specify
+the exact method), and aggregated as needed (e.g. a histogram of all latency values).
+* Accessing the aggregated data. This can be filtered by type of data, name of resources used, 
```

Comment:
s/type of data/data type
s/name of resources used/resource name

Reply:
Done.